modules/openapi-generator-maven-plugin/src/main/java/org/openapitools/codegen/plugin/CodeGenMojo.java

```diff
@@ -758,7 +758,10 @@ public void execute() throws MojoExecutionException {
 
             if (storedInputSpecHashFile.getParent() != null && !new File(storedInputSpecHashFile.getParent()).exists()) {
                 File parent = new File(storedInputSpecHashFile.getParent());
-                parent.mkdirs();
+                if (!parent.mkdirs()) {
+                    throw new RuntimeException("Failed to create the folder " + parent.getAbsolutePath() +
+                                               " to store the checksum of the input spec.");
+                }
```

Comment:
Having a hard time spotting where the temporary directory comes from that could sink to this location. Can you point out the line?

Reply:
The change simply checks to ensure the directory is created successfully. This is not to say without these lines, there's a security issue.

Reply:
Makes sense!